### PR TITLE
Feature/BDD-CODE-IP2-184-registration-form

### PR DESCRIPTION
**Pull request by kmiecikmarcin**

### CODE REVIEW

File: RestSharpProject/Steps/FormModuleSteps.cs [hunk 1]
```diff
@@ -0,0 +1,582 @@
...
+        }
+        
+        [Then(@"JSON body with message about empty field Adres e-mail")]
+        [Obsolete]
+        public void ThenJSONBodyWithMessageAboutEmptyFieldAdresE_Mail()
+        {
+            var responseContent = ScenarioContext.Current["response"].ToString();
+            JObject parseRestResponse = JObject.Parse(responseContent);
+
+            var takeErrorFromResponse = parseRestResponse["fields"]["email"][0].ToString();
```
Comment (hyurii) [inline RestSharpProject/Steps/FormModuleSteps.cs:369]: you should use User model here

File: RestSharpProject/Steps/FormModuleSteps.cs [hunk 2]
```diff
@@ -0,0 +1,582 @@
...
+        
+        [Given(@"User filled required data with checking all technology groups")]
+        [Obsolete]
+        public void GivenUserFilledRequiredDataWithCheckingAllTechnologyGroups()
+        {
+            string[] technologies = new string[4] { "JS", "Java", "QA", "Mobile" };
+
+            user = new UserData("Pani", "Dorota", "Kowalska", email, phoneNumber, technologies, password, login, githubLink);
+
+            ScenarioContext.Current.Add("user", user);
```
Comment (hyurii) [inline RestSharpProject/Steps/FormModuleSteps.cs:125]: are you really need to use ScenarioContext? simple field is not enough for user and response?

**Review by hyurii — CHANGES_REQUESTED** [review]

**Review by katkoch — APPROVED** [review]

**Review by js40598 — APPROVED** [review]

File: RestSharpProject/Steps/RegistrationFormSteps.cs [hunk 3]
```diff
@@ -0,0 +1,472 @@
...
+    public class RegistrationFormSteps
+    {
+        RestClient restClient = new RestClient();
+        RestRequest restRequest = new RestRequest("http://127.0.0.1:8080/api/register", Method.POST);
+        RestResponse restResponse;
+        User user;
+        UserModel userModel;
+        Response responseMessage;
+
+        string email = User.GenerateEmailAdress();
```
Comment (hyurii) [inline RestSharpProject/Steps/RegistrationFormSteps.cs:21]: this should be part of domain model as one method CreateUser()

File: RestSharpProject/Steps/RegistrationFormSteps.cs [hunk 4]
```diff
@@ -0,0 +1,472 @@
...
+        string incorrectPassword = "wrongpassword";
+        string incorrectGithubLink = "wrongLink";
+        string tooShortLogin = "l";
+        string tooLongLogin = "zaDlugiLoginUzytkownika";
+
+        [Given(@"User filled data correctly")]
+        public void GivenUserFilledJanKowalskiExampleEmail_ComQARandomPasswordRandomLoginGithub_ComExampleCorrectly()
+        {
+            List<string> technologies = new List<string>();
+            technologies.Add("QA");
```
Comment (hyurii) [inline RestSharpProject/Steps/RegistrationFormSteps.cs:39]: Enum fits here better in my opinion

File: RestSharpProject/Features/RegistrationForm.feature [hunk 5]
```diff
@@ -0,0 +1,125 @@
+Feature: Registration form
+	#Task in QA Team: https://tracker.intive.com/jira/browse/IP2-182
+	#Task in JS Team: https://tracker.intive.com/jira/browse/IP2-244
+
+#https://tracker.intive.com/jira/browse/IP2-314
```
Comment (hyurii) [inline RestSharpProject/Features/RegistrationForm.feature:5]: Im struggling that we dont have step to check if post rly register user. for example Get , but maybe in the future we can update it

**Review by hyurii — APPROVED** [review]
you should not put generated files into repo. ex * feature.cs

**Pull request merged**